src/IO/S3/Credentials.h

```diff
@@ -1,5 +1,8 @@
 #pragma once
 
+#include <exception>
+#include <variant>
+#include "base/types.h"
```

Comment:
```suggestion
#include <base/types.h>
```
also seems like this include can be removed